clients/rust/tests/harvest_sync_rewards.rs

```diff
@@ -197,6 +197,176 @@ async fn harvest_sync_rewards() {
+        }),
+    );
+
+    let sync_ix = HarvestSyncRewardsBuilder::new()
+        .config(config_manager.config)
+        .sol_staker_stake(sol_staker_stake_manager.stake)
+        .validator_stake(validator_stake_manager.stake)
+        .sol_stake(sol_staker_stake_manager.sol_stake)
+        .destination(destination)
+        .sol_stake_view_program(paladin_sol_stake_view_program_client::ID)
+        .instruction();
+
+    let tx = Transaction::new_signed_with_payer(
+        &[sync_ix],
+        Some(&context.payer.pubkey()),
+        &[&context.payer],
+        context.last_blockhash,
+    );
+    context.banks_client.process_transaction(tx).await.unwrap();
+
+    // Then the destination account has the sync rewards.
+
+    let account = get_account!(context, destination);
+    assert_eq!(account.lamports, 100_000_000 + 1_000_000); // rent + rewards
+
+    // Then the SOL amounts are correctly synced (0 SOL staked).
+
+    let account = get_account!(context, validator_stake_manager.stake);
+    let validator_stake_account = ValidatorStake::from_bytes(account.data.as_ref()).unwrap();
+    assert_eq!(validator_stake_account.total_staked_lamports_amount, 0);
+
+    let account = get_account!(context, sol_staker_stake_manager.stake);
+    let stake_account = SolStakerStake::from_bytes(account.data.as_ref()).unwrap();
+
+    assert_eq!(stake_account.lamports_amount, 0);
+
+    // And the last seen stake rewards per token on the SOL staker stake account
+    // was updated correctly.
+
+    let last_seen_stake_rewards_per_token =
+        stake_account.delegation.last_seen_stake_rewards_per_token;
+
+    let account = get_account!(context, config_manager.config);
+    let config_account = Config::from_bytes(account.data.as_ref()).unwrap();
+
+    assert!(config_account.accumulated_stake_rewards_per_token > last_seen_stake_rewards_per_token);
+    assert!(last_seen_stake_rewards_per_token > 0);
+
+    // Expected rewards per token:
+    //
+    //  - SOL staker stake's token amount: 1_300_000_000
+    //  - total stake rewards available: 1_300_000_000
+    //  - rewards remanining: 1_300_000_000 - 1_000_000 = 1_299_000_000
+    //    (after the searcher rewards are taken)
+    //
+    //  On the SOL staker stake account:
+    //  - last seen rewards per token: 1_000_000 / 1_300_000_000 - 1
+    //
+    //  Effective rewards per token for SOL stake harvest:
+    //  - rewards per token for SOL stake harvest: 1_299_000_000 / 1_300_000_000
+    assert_eq!(
+        last_seen_stake_rewards_per_token,
+        calculate_stake_rewards_per_token(1_000_000, 1_300_000_000) - 1
+    );
+
+    let difference = config_account
+        .accumulated_stake_rewards_per_token
+        .checked_sub(last_seen_stake_rewards_per_token)
+        .and_then(|rewards| {
+            rewards.checked_sub(calculate_stake_rewards_per_token(
+                1_299_000_000,
+                1_300_000_000,
+            ))
+        })
+        .unwrap()
+        - 1;
```

Comment:
micro-nit: the `-1` was a bit confusing, maybe it's clearer as 
```suggestion
    let difference = config_account
        .accumulated_stake_rewards_per_token
        .checked_sub(last_seen_stake_rewards_per_token)
        .and_then(|rewards| {
            rewards.checked_sub(calculate_stake_rewards_per_token(
                1_299_000_000,
                1_300_000_000,
            ) - 1)
        })
        .unwrap();
```

What do you think?

Reply:
Sure, if you think that's more clear I'll add the change.

Reply:
that works, thanks!